fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java

```diff
@@ -1379,6 +1382,21 @@ public void updateLoanApplicationAttributes(JsonCommand command, Loan loan, Map<
             changes.put(LoanProductConstants.IS_EQUAL_AMORTIZATION_PARAM, newValue);
             loanProductRelatedDetail.setEqualAmortization(newValue);
         }
+
+        final boolean isDownPaymentEnabled = validateDownPaymentAttribute(
+                loan.loanProduct().getLoanProductRelatedDetail().isEnableDownPayment(), command.parsedJson());
+        if (isDownPaymentEnabled != loanProductRelatedDetail.isEnableDownPayment()) {
+            changes.put(LoanProductConstants.ENABLE_DOWN_PAYMENT, isDownPaymentEnabled);
+            loanProductRelatedDetail.setEnableDownPayment(isDownPaymentEnabled);
```

Comment:
Its more straightforward to set the values as "false" instead of assigning the value of the `isDownPaymentEnabled` which cannot be anything else but false.

Reply:
We need to consider a couple of scenarios:
1. LP with DownPayment enabled and originally the LA with DownPayment enabled, then LA disable DownPayment, and
2.LP with DownPayment enabled and originally the LA with DownPayment disabled, then LA enable DownPayment

This is why we are validating If we have a change in the new value compared with the old value

Reply:
But then you need to check whether the provided `enableDownPayment` is different than the one is already set on the loan, not on the loan product. 

The following situation will have no effect and the down payment remains disabled at the end.
LP: enable down payment is TRUE
Submit loan: enable down payment is FALSE
Update loan: enabled down payment is TRUE -> Down payment remains disabled hence we are comparing the newly provided value with the LP value.

Reply:
You are checking the loan product configuration, but what you should check is only the configuration on loan. This method is covering the update loan application part, we already have something configured on the loan, thats the only thing that should be checked! 

If on the loan product it was enabled down payment, but it was disabled when the loan was submitted and now when the loan is got updated, we want to enable again, you should not check what was set on the loan product, but check whether on the loan level what was set and now during the update if we wanna change it and enable again, then we can set from the loan product the down payment configurations on the loan level again.